pkg/operation/botanist/controlplane/resourcemanager/resource_manager.go

```diff
@@ -0,0 +1,553 @@
+
+	if err := r.ensureService(ctx); err != nil {
+		return err
+	}
+
+	if err := r.ensureDeployment(ctx); err != nil {
+		return err
+	}
+
+	return r.ensureVPA(ctx)
+}
+
+func (r *ResourceManager) Destroy(ctx context.Context) error {
+	objectsToDelete := []client.Object{
+		r.emptyVPA(),
+		r.emptyDeployment(),
+		r.emptyService(),
+		r.emptyServiceAccount(),
+	}
+	role, err := r.emptyRoleInWatchedNamespace()
+	if err == nil {
+		objectsToDelete = append(objectsToDelete, role)
+	}
+	rb, err := r.emptyRoleBindingInWatchedNamespace()
+	if err == nil {
+		objectsToDelete = append(objectsToDelete, rb)
+	}
+	cr, err := r.emptyClusterRole()
+	if err == nil {
+		objectsToDelete = append(objectsToDelete, cr)
+	}
+	crb, err := r.emptyClusterRoleBinding()
+	if err == nil {
+		objectsToDelete = append(objectsToDelete, crb)
+	}
+
+	return kutil.DeleteObjects(
+		ctx,
+		r.client,
+		objectsToDelete...,
+	)
+}
+
+func (r *ResourceManager) ensureRBAC(ctx context.Context) error {
+	deployResourcesIntoAnotherCluster := r.config.KubeConfig != nil
+	if deployResourcesIntoAnotherCluster {
+		if r.config.WatchedNamespace == nil {
+			if err := r.ensureClusterRole(ctx, allowManagedResources); err != nil {
+				return err
+			}
+			if err := r.ensureClusterRoleBinding(ctx); err != nil {
+				return err
+			}
+		} else {
+			if err := r.ensureRoleInWatchedNamespace(ctx, allowManagedResources); err != nil {
+				return err
+			}
+			if err := r.ensureRoleBinding(ctx); err != nil {
+				return err
+			}
+		}
+	} else {
+		if err := r.ensureClusterRole(ctx, allowAll); err != nil {
+			return err
+		}
+		if err := r.ensureClusterRoleBinding(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ResourceManager) ensureClusterRole(ctx context.Context, policies []rbacv1.PolicyRule) error {
+	clusterRole, err := r.emptyClusterRole()
+	if err != nil {
+		return err
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, r.client, clusterRole, func() error {
+		clusterRole.Labels = *r.config.DefaultLabels
+		clusterRole.Rules = policies
+		return nil
+	})
+	return err
+}
+
+func (r *ResourceManager) emptyClusterRole() (*rbacv1.ClusterRole, error) {
+	if r.config.ClusterRoleName == nil {
+		return nil, errors.New("creating Cluster Role failed - no name defined for the Cluster Role")
+	}
+	return &rbacv1.ClusterRole{ObjectMeta: metav1.ObjectMeta{Name: *r.config.ClusterRoleName}}, nil
+}
+
+func (r *ResourceManager) ensureClusterRoleBinding(ctx context.Context) error {
+	clusterRoleBinding, err := r.emptyClusterRoleBinding()
+	if err != nil {
+		return err
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, r.client, clusterRoleBinding, func() error {
+		clusterRoleBinding.Labels = *r.config.DefaultLabels
+		clusterRoleBinding.RoleRef = rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     *r.config.ClusterRoleName,
+		}
+		clusterRoleBinding.Subjects = []rbacv1.Subject{{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      UserName,
```

Comment:
Given above proposed change of the `UserName`, this won't work anymore in the case that the source cluster = target cluster

Reply:
Yes, I assume that is why we do not change the UserName?

Reply:
OK, see comment above. I guess you are mixing up the service account name and the "user name" (which ends up in the `CommonName` field of a certificate).

Reply:
Yes, that is true. Thanks for pointing this out.